store/localcachelayer/post_layer.go

```diff
@@ -23,26 +35,75 @@ func (s *LocalCachePostStore) handleClusterInvalidateLastPosts(msg *model.Cluste
 }
 
 func (s LocalCachePostStore) ClearCaches() {
+	s.rootStore.doClearCacheCluster(s.rootStore.lastPostTimeCache)
+	s.rootStore.doClearCacheCluster(s.rootStore.postLastPostsCache)
 	s.PostStore.ClearCaches()
 
-	s.rootStore.postLastPostsCache.Purge()
 	if s.rootStore.metrics != nil {
+		s.rootStore.metrics.IncrementMemCacheInvalidationCounter("Last Post Time - Purge")
 		s.rootStore.metrics.IncrementMemCacheInvalidationCounter("Last Posts Cache - Purge")
 	}
 }
 
 func (s LocalCachePostStore) InvalidateLastPostTimeCache(channelId string) {
-	s.PostStore.InvalidateLastPostTimeCache(channelId)
+	s.rootStore.doInvalidateCacheCluster(s.rootStore.lastPostTimeCache, channelId)
 
 	// Keys are "{channelid}{limit}" and caching only occurs on limits of 30 and 60
 	s.rootStore.doInvalidateCacheCluster(s.rootStore.postLastPostsCache, channelId+"30")
 	s.rootStore.doInvalidateCacheCluster(s.rootStore.postLastPostsCache, channelId+"60")
 
+	s.PostStore.InvalidateLastPostTimeCache(channelId)
+
 	if s.rootStore.metrics != nil {
+		s.rootStore.metrics.IncrementMemCacheInvalidationCounter("Last Post Time - Remove by Channel Id")
 		s.rootStore.metrics.IncrementMemCacheInvalidationCounter("Last Posts Cache - Remove by Channel Id")
 	}
 }
 
+func (s LocalCachePostStore) GetEtag(channelId string, allowFromCache bool) string {
+	if allowFromCache {
+		if lastTime := s.rootStore.doStandardReadCache(s.rootStore.lastPostTimeCache, channelId); lastTime != nil {
+			return fmt.Sprintf("%v.%v", model.CurrentVersion, lastTime.(int64))
+		}
+	}
+
+	result := s.PostStore.GetEtag(channelId, allowFromCache)
+
+	splittedResult := strings.Split(result, ".")
+
+	lastTime, _ := strconv.ParseInt((splittedResult[len(splittedResult)-1]), 10, 64)
```

Comment:
This is not the same behavior and I don't like it. Anyway, I thinking we should cache the entire result string, so we don't need to parse and build the string each single time.

Reply:
Other changes corrected. For this, in order to cache the string, we would need a new cache. Is this what we want?
If so, is there any possibility that `model.CurrentVersion` changes during the lifetime of the cache? If so, we would need to invalidate this etag on those cases.

Reply:
No, It is not going to change until you upgrade the server, and in that cases makes a lot of sense to invalidate the cache anyway.

Reply:
I think we should change how we handle the GetEtag from the store entirely, changing the GetEtag calculation outside of the store. That doesn't have anything to do with the store itself, we need to get the data from the store needed to generate the Etag, but not the etag itself. I think we can go forward with this approach and later on change the GetEtag function entirely with something like GetLastPostDate, or something like that.